imap_processing/idex/idex_packet_parser.py

```diff
@@ -0,0 +1,286 @@
+        )
+        # Extract the 17-22-bit integer (usually 8)
+        lspretrigblocks = (pkt.data["IDX__TXHDRBLOCKS"].derived_value >> 16) & 0b1111
+        # Extract the next 4-bit integer (usually 8)
+        lsposttrigblocks = (pkt.data["IDX__TXHDRBLOCKS"].derived_value >> 12) & 0b1111
+        # Extract the next 6 bits integer (usually 32)
+        hspretrigblocks = (pkt.data["IDX__TXHDRBLOCKS"].derived_value >> 6) & 0b111111
+        # Extract the first 6 bits (usually 32)
+        hsposttrigblocks = (pkt.data["IDX__TXHDRBLOCKS"].derived_value) & 0b111111
+        logging.info("HS pre trig sampling blocks: " + str(hspretrigblocks))
+        logging.info("LS pre trig sampling blocks: " + str(lspretrigblocks))
+        logging.info("HS post trig sampling blocks: " + str(hsposttrigblocks))
+        logging.info("LS post trig sampling blocks: " + str(lsposttrigblocks))
+        tof_delay = (
+            pkt.data["IDX__TXHDRSAMPDELAY"].raw_value >> 2
+        )  # First two bits are padding
+        mask = 0b1111111111
+        lgdelay = (tof_delay) & mask
+        mgdelay = (tof_delay >> 10) & mask
+        hgdelay = (tof_delay >> 20) & mask
+        logging.info(f"High gain delay = {hgdelay} samples.")
+        logging.info(f"Mid gain delay = {mgdelay} samples.")
+        logging.info(f"Low gain delay = {lgdelay} samples.")
+        if (
+            pkt.data["IDX__TXHDRLSTRIGMODE"].derived_value != 0
+        ):  # If this was a LS (Target Low Gain) trigger
+            self.Triggerorigin = "LS"
+            logging.info("Low sampling trigger mode enabled.")
+        logging.info(
+            "Packet low trigger mode = "
+            + str(pkt.data["IDX__TXHDRLGTRIGMODE"].derived_value)
+        )
+        logging.info(
+            "Packet mid trigger mode = "
+            + str(pkt.data["IDX__TXHDRMGTRIGMODE"].derived_value)
+        )
+        logging.info(
+            "Packet high trigger mode = "
+            + str(pkt.data["IDX__TXHDRHGTRIGMODE"].derived_value)
+        )
+        if pkt.data["IDX__TXHDRLGTRIGMODE"].derived_value != 0:
+            logging.info("Low gain TOF trigger mode enabled.")
+            self.Triggerorigin = "LG"
+        if pkt.data["IDX__TXHDRMGTRIGMODE"].derived_value != 0:
+            logging.info("Mid gain TOF trigger mode enabled.")
+            self.Triggerorigin = "MG"
+        if pkt.data["IDX__TXHDRHGTRIGMODE"].derived_value != 0:
+            logging.info("High gain trigger mode enabled.")
+            self.Triggerorigin = "HG"
+        logging.info(
+            f"AID = {pkt.data['IDX__SCI0AID'].derived_value}"
+        )  # Instrument event number
+        logging.info(
+            f"Event number = {pkt.data['IDX__SCI0EVTNUM'].raw_value}"
+        )  # Event number out of how many events constitute the file
+        logging.info(
+            f"Rice compression enabled = {bool(pkt.data['IDX__SCI0COMP'].raw_value)}"
+        )
+
+
+class IDEXRawDustEvent:
```

Comment:
I'm not sure how we feel about naming classes. I guess I'd vote for dropping IDEX personally. So that if I were to import it later it'd be: `from imap_processing.idex import RawDustEvent` and not duplicating IDEX. But, I don't have a strong preference.